### PR TITLE
fix lint errors, add tests

### PR DESCRIPTION
**Pull request by flynneva**
Signed-off-by: Evan Flynn <evan.flynn@apex.ai>

---
This change is [Reviewable](https://reviewable.io/reviews/apexai/reference-system-autoware/6)

### CODE REVIEW

File: reference_system_autoware/CMakeLists.txt [hunk 1]
```diff
@@ -30,23 +30,35 @@ ament_auto_add_executable(autoware_default_multithreaded
...
-  #      ${CMAKE_CURRENT_SOURCE_DIR}/results/plot.png
-  #      5
-  #    ARG
-  #    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  #    COMMENT "Run psrecord for specified target")
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+
+  # run target for 5 seconds
+  function(test_reference_system_autoware_5s target)
```
Comment (flynneva) [inline reference_system_autoware/CMakeLists.txt:39]: this function should allow us to pass in different executables and run the same unit tests on them i think

File: reference_system_autoware/CMakeLists.txt [hunk 2]
```diff
@@ -30,23 +30,35 @@ ament_auto_add_executable(autoware_default_multithreaded
...
+        "rclcpp" "reference_interfaces")
+    endif()
+  endfunction()
+
+  # get available rmw implementations
+  find_package(rmw_implementation_cmake REQUIRED)
+  get_available_rmw_implementations(rmws_available)
+
+  # loop over each rmw implmentation
+  foreach(rmw_implementation ${rmws_available})
```
Comment (flynneva) [inline reference_system_autoware/CMakeLists.txt:68]: this should let us auto-run on different rmw's too if we want to eventually break that out as well in the results

File: reference_system_autoware/CMakeLists.txt [hunk 3]
```diff
@@ -30,23 +30,35 @@ ament_auto_add_executable(autoware_default_multithreaded
...
+    find_package("${rmw_implementation}" REQUIRED)
+
+    # add each exe to test
+    test_reference_system_autoware_5s(autoware_default_multithreaded)
```
Comment (flynneva) [inline reference_system_autoware/CMakeLists.txt:72]: here is where we actually run the function we defined above and pass in the different executables we will want to run the tests on.

**Comment by flynneva** [comment]
@landerU you might find this interesting too since you added the CI to the main repo...for now there is just a dummy test in there until we add more.

what im thinking is we run the exe thats passed into the CMake function for some time limit and we run some tests that check number of nodes, publishers, subscribers, message size, etc. to make sure it matches whats defined in the [requirements for the reference system](https://github.com/ApexAI/reference-system-autoware/tree/feature/add_requirements/reference_system_autoware#reference-system-autoware)

**Comment by LanderU** [comment]
@flynneva, why don't we use ros_lauch for that?

**Review by elfenpiff — APPROVED** [review]

**Pull request merged**